File: app/src/main/java/com/app/l_pesa/common/CommonEditTextRegular.java
package com.app.l_pesa.common;

import android.content.Context;
import android.graphics.Typeface;
import androidx.appcompat.widget.AppCompatEditText;
import android.text.InputFilter;
import android.text.Spanned;
import android.util.AttributeSet;


public class CommonEditTextRegular extends AppCompatEditText
{
    public CommonEditTextRegular(Context context) {
        super(context);
        init(context);
    }

    public CommonEditTextRegular(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public CommonEditTextRegular(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(context);
        setFilters(new InputFilter[]{new EmojiFilter()});
    }

    private void init(Context mContext) {

            Typeface face = Typeface.createFromAsset(mContext.getAssets(), "fonts/Montserrat-Regular.ttf");
            this.setTypeface(face);

    }

    private class EmojiFilter implements InputFilter {

        @Override
        public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
            for (int i = start; i < end; i++) {
                int type = Character.getType(source.charAt(i));
                if (type == Character.SURROGATE || type == Character.OTHER_SYMBOL) {
                    return "";
                }
            }
            return null;
        }
    }
}

File: app/src/main/java/com/app/l_pesa/common/CommonEditTextRegular.kt
package com.app.l_pesa.common

import android.content.Context
import android.graphics.Typeface
import android.text.InputFilter
import android.text.Spanned
import android.util.AttributeSet

import androidx.appcompat.widget.AppCompatEditText


class CommonEditTextRegular : AppCompatEditText {
    constructor(context: Context) : super(context) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init(context)
        filters = arrayOf<InputFilter>(EmojiFilter())
    }

    private fun init(mContext: Context) {

        val face = Typeface.createFromAsset(mContext.assets, "fonts/Montserrat-Regular.ttf")
        this.typeface = face

    }

    private inner class EmojiFilter : InputFilter {

        override fun filter(source: CharSequence, start: Int, end: Int, dest: Spanned, dstart: Int, dend: Int): CharSequence? {
            for (i in start until end) {
                val type = Character.getType(source[i])
                if (type == Character.SURROGATE.toInt() || type == Character.OTHER_SYMBOL.toInt()) {
                    return ""
                }
            }
            return null
        }
    }
}

